### PR TITLE
Find command with prefix v1.3

### LINKED ISSUE #63: The "Find" command does not work with subjects
There is currently no support for finding a student based on the subjects they take.

### PR DESCRIPTION
**Pull request by joosxi**
Allow find command to work with prefixes for name (n/), class (c/) and subject (s/).

### CODE REVIEW

File: src/main/java/seedu/address/model/person/FindCommandPredicate.java [hunk 1]
```diff
@@ -18,18 +21,58 @@ public FindCommandPredicate(List<String> keywords) {
...
+     */
+    private boolean stringContainsWord(String sentence, String keywords) {
+        String[] keywordArr = keywords.split(" ");
+        for (String keyword: keywordArr) {
+            if (StringUtil.containsWordIgnoreCase(sentence, keyword)) {
+                return true;
+            }
+        }
+        return false;
+    }
```
Comment (e1010101) [inline src/main/java/seedu/address/model/person/FindCommandPredicate.java:51]: Good javadoc here + in other places!

File: src/main/java/seedu/address/model/person/FindCommandPredicate.java [hunk 2]
```diff
@@ -18,18 +21,58 @@ public FindCommandPredicate(List<String> keywords) {
...
+        String subjectsString = String.join(" ", subjectsSet);
+
+        String[] keywordArr = keywords.split(" ");
+        for (String keyword: keywordArr) {
+            if (StringUtil.containsWordIgnoreCase(subjectsString, keyword)) {
+                return true;
+            }
+        }
+        return false;
+    }
```
Comment (e1010101) [inline src/main/java/seedu/address/model/person/FindCommandPredicate.java]: Can this method be migrated to SubjectHandler?

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 3]
```diff
@@ -31,33 +34,33 @@ public class FindCommandParser implements Parser<FindNameCommand> {
      */
     public FindNameCommand parse(String args) throws ParseException {
         String trimmedArgs = args.trim();
-        Name foundName = DEFAULT_NAME;
-        StudentClass foundClass = DEFAULT_CLASS;
-        Subject foundSubjects = DEFAULT_SUBJECT;
+        List<String> findCommandKeywords = new ArrayList<String>();
```
Comment (e1010101) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java]: Good implementation of multiple keyword parsing!

**Review by e1010101 — CHANGES_REQUESTED** [review]
Changes requested as per comments! Otherwise, LGTM

**Review by e1010101 — APPROVED** [review]
LGTM!

**Review by jgyj123 — APPROVED** [review]
LGTM 👍

**Pull request merged**